recipes-devtools/opkg/opkg-keyrings_%.bbappend

```diff
@@ -1,6 +1,14 @@
 FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
 
-PV = "${DISTRO_VERSION}"
+python () {
+	buildname = d.getVar('BUILDNAME')
+	import re
+	# Allow for BUILDNAME to be in the form of major.Minor or major.Minor.Update
+	package_version = re.match(r'(\d+\.\d+\.\d+|\d+\.\d+)', buildname)
+	if match:
+		buildname = package_version.group(1)
+	d.setVar('PV', buildname)
```

Comment:
Is this local to this recipe?

Reply:
PKGV seems to be what other recipes wanting dynamic PV are using.

E.g., https://github.com/ni/openembedded-core/blob/nilrt/master/next/meta/recipes-kernel/perf/perf.bb#L382
https://github.com/ni/meta-openembedded/blob/nilrt/master/next/meta-oe/recipes-kernel/cpupower/cpupower.bb#L32
https://github.com/ni/meta-nilrt/blob/nilrt/master/next/recipes-core/dist-nilrt/dist-nilrt-grub-gateway.bb#L38

Many other examples in codesearch.